src/braket/circuits/gates.py

```diff
@@ -1091,6 +1091,118 @@ def phaseshift(
 Gate.register_gate(PhaseShift)
 
 
+class U(TripleAngledGate):
+    """Parameterized primitive gate for OpenQASM simulator
+
+    Args:
+        angle_1 (Union[FreeParameterExpression, float]): theta angle in radians.
+        angle_2 (Union[FreeParameterExpression, float]): phi angle in radians.
+        angle_3 (Union[FreeParameterExpression, float]): lambda angle in radians.
+    """
+
+    def __init__(
+        self,
+        angle_1: Union[FreeParameterExpression, float],
+        angle_2: Union[FreeParameterExpression, float],
+        angle_3: Union[FreeParameterExpression, float],
+    ):
+        super().__init__(
+            angle_1=angle_1,
+            angle_2=angle_2,
+            angle_3=angle_3,
+            qubit_count=None,
+            ascii_symbols=[_multi_angled_ascii_characters("U", angle_1, angle_2, angle_3)],
+        )
+
+    @property
+    def _qasm_name(self) -> str:
+        return "U"
+
+    def to_matrix(self) -> np.ndarray:
+        """
+        Generate parameterized Unitary matrix.
+        https://openqasm.com/language/gates.html#built-in-gates
+
+        Returns:
+            np.ndarray: U Matrix
+        """
+        theta = self.angle_1
+        phi = self.angle_2
+        lam = self.angle_3
+        return np.array(
+            [
+                [
+                    np.cos(theta / 2),
+                    -np.exp(1j * lam) * np.sin(theta / 2),
+                ],
+                [
+                    np.exp(1j * phi) * np.sin(theta / 2),
+                    np.exp(1j * (phi + lam)) * np.cos(theta / 2),
+                ],
+            ]
+        )
+
+    def adjoint(self) -> list[Gate]:
+        return [U(self.angle_1, np.pi - self.angle_3, np.pi - self.angle_2)]
```

Comment:
Is this equivalent to the [mapping stated in the docs](https://openqasm.com/language/gates.html#:~:text=The%20base%20case%20is%20given%20by%20replacing%20inv%20%40%20U(%CE%B8%2C%20%CF%95%2C%20%CE%BB)%20by%20U(%2D%CE%B8%2C%20%2D%CE%BB%2C%20%2D%CF%95))?

Reply:
I don't believe the local simulator defines an adjoint for `U`. The OpenQASM spec states that the inverse of `U` can be expressed as `U(self.angle_1, -self.angle_3, -self.angle_2)`, which seems simpler here. Can we confirm they're equivalent and opt for the simpler expression?